### PR TITLE
Add more basic explanation to case-when documentation and indicate order of evaluation

### PR DESCRIPTION
**Pull request by jpeacock29**

### CODE REVIEW

File: R/case-when.R [hunk 1]
```diff
@@ -146,7 +147,7 @@
 #'   mutate(type = case_character_type(height, mass, species, robots = FALSE)) %>%
 #'   pull(type)
 case_when <- function(...,
-                      .default = NULL,
+                      .default = NA,
```
Comment (hadley) [inline R/case-when.R:150]: I changed the default to `NA` here — this seems intuitively correct to me, doesn't break any unit tests, and IMO removes one layer of indirection.

Comment (DavisVaughan) [reply R/case-when.R]: I think we should consider this in a broader issue, as it also affects `case_match()`, `vec_case_when()`, and `vec_case_match()`. I'm also not sure I agree.

I've reverted for now just so we can more easily merge this, but I'll open an issue

**Comment by DavisVaughan** [comment]
Thanks @jpeacock29!

**Comment by jpeacock29** [comment]
Cheers, thank you as well!

**Pull request merged**